lib/scenario.js

```diff
@@ -54,7 +54,7 @@ module.exports.test = (test) => {
 
     if (isAsyncFunction(testFn)) {
       event.emit(event.test.started, test);
-      testFn.call(test, getInjectedArguments(testFn, test)).then(() => {
+      Promise.resolve().then(() => testFn.call(test, getInjectedArguments(testFn, test))).then(() => {
```

Comment:
I don't think this is needed testFn is AsyncFunction
by definition async function returns a promise
so you dont need to start a new promise here

Reply:
By promisifying this call we can catch any error from **getInjectedArguments**, which is core of my problem. Instead of promisifying `testFn.call` of course we can wrap **getInjectedArguments** in try / catch block, but this is isconsistent behavior. If we do that, we also should throw error in another `recorder.throw` / `recorder.catch` block, and I dont think this is ok: we already have `recorder.throw` / `recorder.catch` block in `.catch`.

Btw, few strings down we are wrapping all `testFn.call` in try / catch with **getInjectedArguments**, so why we cant promisify it here?

Reply:
Added description!